framework/web/UrlRule.php

```diff
@@ -367,8 +367,15 @@ public function createUrl($manager, $route, $params)
                 continue;
             }
             if (!isset($params[$name])) {
-                return false;
-            } elseif (strcmp($params[$name], $value) === 0) { // strcmp will do string conversion automatically
+                // allow omit empty optional params
+                // @see https://github.com/yiisoft/yii2/issues/10970
+                if (in_array($name, $this->placeholders) && strcmp($value, '') === 0) {
```

Comment:
@liongen If you want add default language in this way, you should override default `UrlManager::createUrl()` and inject default language into params before URL creation. Magic injecting the default parameters when creating URLs will create unexpected restults and probably will break BC.